src/GraphBLAS-sharp.Backend/Matrix/CSRMatrix/Elementwise.fs

```diff
@@ -0,0 +1,774 @@
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let leftMergedValues =
+                clContext.CreateClArray<'a>(
+                    resLength,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let rightMergedValues =
+                clContext.CreateClArray<'b>(
+                    resLength,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let isEndOfRow =
+                clContext.CreateClArray<int>(
+                    resLength,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let isLeft =
+                clContext.CreateClArray<int>(
+                    resLength,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let ndRange =
+                Range1D.CreateValid((matrixLeftRowPointers.Length - 1) * workGroupSize, workGroupSize)
+
+            let kernel = kernel.GetKernel()
+
+            processor.Post(
+                Msg.MsgSetArguments
+                    (fun () ->
+                        kernel.KernelFunc
+                            ndRange
+                            (matrixLeftRowPointers.Length - 1)
+                            matrixLeftRowPointers
+                            matrixLeftColumns
+                            matrixLeftValues
+                            matrixRightRowPointers
+                            matrixRightColumns
+                            matrixRightValues
+                            allRows
+                            allColumns
+                            leftMergedValues
+                            rightMergedValues
+                            isEndOfRow
+                            isLeft)
+            )
+
+            processor.Post(Msg.CreateRunMsg<_, _>(kernel))
+
+            allRows, allColumns, leftMergedValues, rightMergedValues, isEndOfRow, isLeft
+
+    let elementwise<'a, 'b, 'c when 'a: struct and 'b: struct and 'c: struct and 'c: equality>
+        (clContext: ClContext)
+        (opAdd: Expr<'a option -> 'b option -> 'c option>)
+        workGroupSize
+        =
+
+        let merge = merge clContext workGroupSize
+
+        let preparePositions =
+            preparePositions clContext opAdd Utils.defaultWorkGroupSize
```

Comment:
All steps of the elementwise except merge are working with the best performance for a relatively large workGroupSize regardless of the characteristics of the matrices. On the other hand, the choice of the best workGroupSize for merge may depend on matrices and it makes sense to vary it. It is possible to add a second parameter for workGroupSize, but since its choice is not so important, it was decided to apply the standard size.